core/bulkchange.class.inc.php

```diff
@@ -567,12 +640,94 @@ protected function PrepareObject(&$oTargetObj, $aRowData, &$aErrors)
 		}
 		return $aResults;
 	}
-	
+
+	/**
+	 * @param $sTargetClass: target class of the failed search
+	 * @param $sForeignAttCode : foreign attribute code of the foreign search
+	 * @param $value : unfound value
+	 *
+	 * @return \CellStatus_SearchIssue
+	 * @throws \CoreException
+	 * @throws \MissingQueryArgument
+	 * @throws \MySQLException
+	 * @throws \MySQLHasGoneAwayException
+	 */
+	protected function GetCellSearchIssue($sTargetClass, $sForeignAttCode, $value) : CellStatus_SearchIssue {
+		//current search with current permissions did not match
+		//let's search why and give some more feedbackst to the user
+
+		//count all objects with all permissions
+		$oReconFilter2 = new DBObjectSearch($sTargetClass);
+		$oReconFilter2->AllowAllData(true);
+		$oExtObjectSet = new CMDBObjectSet($oReconFilter2);
+		$iAllowAllDataObjectCount = $oExtObjectSet->Count();
+
+		//count all objects with current user permissions
+		$oReconFilter2->AllowAllData(false);
+		$oExtObjectSetWithCurrentUserPermissions = new CMDBObjectSet($oReconFilter2);
+		$iCurrentUserRightsObjectCount = $oExtObjectSetWithCurrentUserPermissions->Count();
+
+		if ($iAllowAllDataObjectCount === 0) {
+			//no objects at all
+			$sReason = Dict::Format('UI:CSVReport-Value-NoMatch-NoObject', $sTargetClass);
+			return new CellStatus_SearchIssue($sReason);
+
+			//search link for exhaustive values
+		}
+
+		if ($iCurrentUserRightsObjectCount === 0){
+			//no objects visible by current user
+			$sReason = Dict::Format('UI:CSVReport-Value-NoMatch-NoObject-ForCurrentUser',
+				$sTargetClass);
```

Comment:
```suggestion
			// No objects visible by current user
			$sReason = Dict::Format('UI:CSVReport-Value-NoMatch-NoObject-ForCurrentUser', $sTargetClass);
```

_We don't wrap before 240 in code conventions_